torchdata/datapipes/iter/util/combining.py

```diff
@@ -213,3 +216,139 @@ def __iter__(self) -> Iterator:
 
     def __len__(self) -> int:
         return len(self.source_iterdatapipe)
+
+
+def _drop_index(idx_data):
+    idx, data = idx_data
+    return data
+
+
+@functional_datapipe("round_robin_demux")
+class RoundRobinDemultiplexerIterDataPipe(IterDataPipe):
```

Comment:
curious: why `demux` is PyTorch core (https://github.com/pytorch/pytorch/blob/7a2930b357a4e62bb0bab53bb0d23c607b6ede38/torch/utils/data/datapipes/iter/combining.py#L305-L306) but `round_robin_demux` is in TorchData?

Reply:
Historical reason. We put lots of DataPipe in core but we plan to move them to torchdata eventually. To reduce the amount of work, I think it's better to put `round_robin_demux` to torchdata directly.